tests/installation/zdup_prepare_repos.pm

```diff
@@ -0,0 +1,103 @@
+
+    # This is just for reference to know how the network was configured prior to the update
+    script_run "ip addr show";
+    save_screenshot;
+
+    # before disable we need to have cdrkit installed to get proper iso appid
+    script_run "zypper -n in cdrkit-cdrtools-compat";
+    # Disable all repos, so we do not need to remove one by one
+    # beware PackageKit!
+    script_run("zypper modifyrepo --all --disable | tee /dev/$serialdev", 0);
+    my $out = wait_serial([$zypper_packagekit, $zypper_repo_disabled], 120);
+    while ($out) {
+        if ($out =~ $zypper_packagekit || $out =~ $zypper_packagekit_again) {
+            send_key 'y';
+            send_key 'ret';
+        }
+        elsif ($out =~ $zypper_repo_disabled) {
+            last;
+        }
+        $out = wait_serial([$zypper_repo_disabled, $zypper_packagekit_again, $zypper_packagekit], 120);
+    }
+    unless ($out) {
+        save_screenshot;
+        $self->result('fail');
+        return;
+    }
+
+    my $defaultrepo;
+    if (get_var('SUSEMIRROR')) {
+        $defaultrepo = "http://" . get_var("SUSEMIRROR");
+    }
+    else {
+        #SUSEMIRROR not set, zdup from ftp source for online migration
+        if (get_var('TEST') =~ /migration_zdup_online_sle12_ga/) {
+            my $flavor  = get_var("FLAVOR");
+            my $version = get_var("VERSION");
+            my $build   = get_var("BUILD");
+            my $arch    = get_var("ARCH");
+            $defaultrepo = "$utils::OPENQA_FTP_URL/SLE-$version-$flavor-$arch-Build$build-Media1";
+        }
+        else {
+            # SUSEMIRROR not set, zdup from attached ISO
+            my $build  = get_var("BUILD");
+            my $flavor = get_var("FLAVOR");
+            script_run "ls -al /dev/disk/by-label";
+            my $isoinfo = "isoinfo -d -i /dev/\$dev | grep \"Application id\" | awk -F \" \" '{print \$3}'";
+
+            script_run "dev=;
+                       for i in sr0 sr1 sr2 sr3 sr4 sr5; do
+                       label=`$isoinfo`
+                       case \$label in
+                           *$flavor-*$build*) echo \"\$i match\"; dev=\"/dev/\$i\"; break;;
+                           *) continue;;
+                       esac
+                       done
+                       [ -z \$dev ] || echo \"found dev \$dev with label \$label\"";
+            # get all attached ISOs including addons' as zdup dup repos
+            my $srx = script_output("ls -al /dev/disk/by-label | grep -E /sr[0-9]+ | wc -l");
+            for my $n (0 .. $srx - 1) {
+                $defaultrepo .= "dvd:/?devices=\${dev:-/dev/sr$n}+";
+            }
+        }
+    }
+
+    my $nr = 1;
+    foreach my $r (split(/\+/, get_var("ZDUPREPOS", $defaultrepo))) {
+        assert_script_run("zypper -n addrepo \"$r\" repo$nr");
+        $nr++;
+    }
+    assert_script_run("zypper -n refresh", 240);
+}
+
+sub test_flags {
+    return {fatal => 1};
```

Comment:
This flag is already set in `installbasetest`, no need to override method.

Reply:
Yes, I removed the flag in  tests/installation/zdup_prepare_repos.pm  and tests/console/zdup_prepare_leap_sle.pm. Thanks.